src/utils.c

```diff
@@ -78,15 +78,21 @@ uint64_t avifNTOH64(uint64_t l)
 
 AVIF_ARRAY_DECLARE(avifArrayInternal, uint8_t, ptr);
 
-void avifArrayCreate(void * arrayStruct, uint32_t elementSize, uint32_t initialCapacity)
+// On error, must set arr->ptr to NULL and both arr->count and arr->capacity to 0.
+avifResult avifArrayCreate(void * arrayStruct, uint32_t elementSize, uint32_t initialCapacity)
```

Comment:
Are you planning to automate (e.g. through a macro) the error code forwarding? Because currently the status is always compared to OK and the value itself never transmitted up the call stack. Otherwise a bool would be enough (which seems traditional to me for functions that "just" allocate memory).

Reply:
Good question. This function is an internal function. We can make sure it never receives invalid arguments, so the only reason for failure is the out-of-memory error.

I am indeed debating whether this function should return avifBool instead. Is that also your preference?

Reply:
Good question. This function is an internal function. We can make sure it never receives invalid arguments, so it can only fail because it runs out of memory.

I am indeed debating whether this function should return `avifBool` instead. Is that also your preference?

Reply:
If there is a global AVIF_CHECK_STATUS macro as mentioned above, I would tend to return a status to leverage it. Otherwise I do not have a strong preference, maybe slightly towards an avifBool to indicate that the only possible error is a memory one indeed. Up to you.

Reply:
OK, I will change `avifArrayCreate()` to return `avifBool`.